galaxy/main/models/collection.py

```diff
@@ -64,9 +65,20 @@ class Meta:
 
     @property
     def latest_version(self):
-        return CollectionVersion.objects.filter(
-            collection=self
-        ).latest('pk')
+        versions = self.versions.filter(hidden=False)
+        if not versions:
+            return None
+        return versions.latest('pk')
+
+    @property
+    def highest_version(self):
+        versions = self.versions.filter(hidden=False)
+        if not versions:
+            return None
+
+        d = {semantic_version.Version(v.version): v for v in versions}
+        highest = semantic_version.Spec('*').select(d.keys())
```

Comment:
@alikins any particular reason you wanted to use `semantic_version.Spec('*')` instead  of simple comparison, which does literally the same?

Reply:
@cutwater Using a semantic_version.Spec() provides a way to filter or blacklist if we need to at some point. Also allows the use of Spec.select() which does the ordering itself.

We don't necessarily need the filter/blacklist support now, so it may be YAGNI.

Reply:
Eventually this property should be replaced with FK, to calculate it once during import, shouldn't it?

Reply:
A `ForeignKey` reference might be the better way to do it, however if that particular `CollectionVersion` was flagged as hidden, we may need to update the `ForeignKey`